src/vs/workbench/contrib/terminal/node/terminalProfiles.ts

```diff
@@ -198,15 +198,31 @@ async function getWslProfiles(wslPath: string, detectWslProfiles?: boolean, logS
 	return [];
 }
 
-async function detectAvailableUnixProfiles(): Promise<ITerminalProfile[]> {
+async function detectAvailableUnixProfiles(quickLaunchOnly?: boolean, configProfiles?: any): Promise<ITerminalProfile[]> {
 	const contents = await fs.promises.readFile('/etc/shells', 'utf8');
 	const profiles = contents.split('\n').filter(e => e.trim().indexOf('#') !== 0 && e.trim().length > 0);
-	return profiles.map(e => {
+	const detectedProfiles = await profiles.map(e => {
 		return {
 			profileName: basename(e),
 			path: e
 		};
 	});
+	if (!quickLaunchOnly) {
+		return detectedProfiles;
+	}
+	let validProfiles: ITerminalProfile[] = [];
+
+	for (const [profileKey, value] of Object.entries(configProfiles)) {
```

Comment:
I think the use of `Object.entries` here is why `Map` is working:

<img width="215" alt="Screen Shot 2021-03-17 at 4 11 16 PM" src="https://user-images.githubusercontent.com/2193314/111550473-6909c400-873b-11eb-98ef-31aeb47534f1.png">

The type isn't a `Map` as this wouldn't work:

<img width="171" alt="Screen Shot 2021-03-17 at 4 11 58 PM" src="https://user-images.githubusercontent.com/2193314/111550511-82127500-873b-11eb-9cdc-2600550748ab.png">

Reply:
ok changed from `Map<string, ITerminalProfileObject>`  to ` { [key: string]: ITerminalProfileObject };`

Reply:
https://github.com/microsoft/vscode/blob/ff1ce589e7336c880152384d5f784dac0410f48b/src/vs/workbench/contrib/terminal/common/terminal.ts#L99-L99